lib/rules/component-data.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @fileoverview When using the data property on a component (i.e. anywhere except on new Vue), the value must be a function that returns an object.
+ * @author Armano
+ */
+'use strict'
+
+const utils = require('../utils')
+
+function create (context) {
+  // variables should be defined here
+
+  // ----------------------------------------------------------------------
+  // Helpers
+  // ----------------------------------------------------------------------
+
+  // any helper functions should go here or else delete this section
+
+  // ----------------------------------------------------------------------
+  // Public
+  // ----------------------------------------------------------------------
+
+  return Object.assign({},
+    utils.executeOnVueComponent(context, properties => {
+      const dataPropertiesNode = properties
+        .filter(p =>
+          p.key.type === 'Identifier' &&
+          p.key.name === 'data' &&
+          p.value.type !== 'FunctionExpression' &&
+          p.value.type !== 'ArrowFunctionExpression'
```

Comment:
I'm not really sure if we should allow to use `ArrowFunctionExpression` in `data` due to changed context of function.